pandas/core/apply.py

```diff
@@ -44,7 +49,38 @@ def frame_apply(
 
 
 class FrameApply:
-    def __init__(self, obj, func, raw, result_type, ignore_failures, args, kwds):
+
+    # ---------------------------------------------------------------
+    # Abstract Methods
+
+    @property
+    def axis(self) -> int:
+        raise AbstractMethodError(self)
```

Comment:
can we not use @abc.abstractmethod going forward for new code?

Reply:
OK.  It looks like for axis I can do just `axis: int` but for the others I have to use property+abc.abstractmethod.

Updated.